operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/event/source/informer/InformerEventSource.java

```diff
@@ -149,16 +159,11 @@ public void onUpdate(R oldObject, R newObject) {
           newObject.getMetadata().getResourceVersion(),
           oldObject.getMetadata().getResourceVersion());
     }
-    primaryToSecondaryIndex.onAddOrUpdate(newObject);
-    onAddOrUpdate(
-        Operation.UPDATE,
-        newObject,
-        oldObject,
-        () -> InformerEventSource.super.onUpdate(oldObject, newObject));
+    onAddOrUpdate(Operation.UPDATE, newObject, oldObject);
   }
 
   @Override
-  public void onDelete(R resource, boolean b) {
+  public synchronized void onDelete(R resource, boolean b) {
```

Comment:
I don't understand why we have this synchronized, could you elaborate pls?

Reply:
> That could also be done with a more granular lock around the usages of primaryToSecondaryIndex.

Yes, that might be cleaner.

> In general there's effectively a single threaded model of event delivery, so unless users are calling the onXXX methods manually, those calls will be serialized. 

Yeah, synchronizing the start with those makes it a bit safer and easier to reason about (without a real performance impact )